### DISCUSSION TITLE
Keyboard shortcut keys for diagram operations

### DISCUSSION THREAD

**Discussion by sathya-1994** [eclipse-glsp/glsp#660] [General]
I am using GLSP's vscode integration project.

My requirement: I want add below shortcut keys for diagram operations
**Refresh:** F5
**Node/Label edit:** F2 (currently we have to double click)

How we can add these shortcut keys?

**Comment by planger** [comment]
Please follow those steps and see how it is done in the example:

1. [Registration of the command](https://github.com/eclipse-glsp/glsp-vscode-integration/blob/4484750c642ccfb0143da69b868237771b1764c6/example/workflow/extension/package.json#L62
) in the extension's `package.json`
2. [Registration of the keybinding](https://github.com/eclipse-glsp/glsp-vscode-integration/blob/4484750c642ccfb0143da69b868237771b1764c6/example/workflow/extension/package.json#L161
) in the extension's `package.json`
3. [Command definition](https://github.com/eclipse-glsp/glsp-vscode-integration/blob/4484750c642ccfb0143da69b868237771b1764c6/example/workflow/extension/src/workflow-extension.ts#L72) for specifying which GLSP action to send for a command in the extension code

The specified action will then be dispatched to the handler that is registered, either on the server or on the client, or both. See the GLSP documentation on [actions and action handlers](https://www.eclipse.org/glsp/documentation/actionhandler/) for more details.
